### PR TITLE
Open API Endpoint for User Data

### PR DESCRIPTION
**Pull request by AKidd95**
#### Description
Changes proposed in this pull request:

* Open API enpoints to request user data from backend
* Test API requests successfully

#### What I have learned working on this feature: 
* how to use texture files
* how to do format request specs
* how to setup and configure devise_token_auth

### CODE REVIEW

File: app/controllers/api/v1/user_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  def show
+    render json: @user, status: :ok
+  end
+
+  private
+
+  def users_near_you
+    @users = User.all
+    @user = User.find_by(id: params[:id])
```
Comment (tochman) [inline app/controllers/api/v1/user_controller.rb]: I’m not sure I understand this

File: config/routes.rb [hunk 2]
```diff
@@ -1,7 +1,13 @@
 Rails.application.routes.draw do
+
   namespace :api do
     namespace :v0 do
       resources :pings, only: [:index], constraints: { format: 'json' }
     end
+
+    namespace :v1 do
+      resources :user, constraints: {format: 'json'}
```
Comment (tochman) [inline config/routes.rb]: Resources are usually set in plural.

File: spec/fixtures/files/user.txt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+{"data" => {
+   "id" => 33,
+   "provider" => "email",
+   "uid" => "hello@itsme.com",
+   "allow_password_change" => false,
+   "name" => "Aiden Jubelin",
+   "nickname" => "MeepMills",
+   "image" => nil,
+   "email" => "hello@itsme.com"
+  }}
```
Comment (tochman) [inline spec/fixtures/files/user.txt]: I note that you are not using the JSON:API serialization standard. Why not?

File: spec/requests/api/v1/user_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::UserController, type: :request do
+  describe 'GET /api/v1/user' do
+    let!(:user) { create(:user) }
+    let(:object) { JSON.parse(response.body)}
+
+    it 'Should return all users' do
+      get '/api/v1/user', params: {
+        email: user.email, encrypted_password: user.encrypted_password
```
Comment (tochman) [inline spec/requests/api/v1/user_request_spec.rb]: How can you send the encrypted password in params?  This don’t roller requires a user to be authenticated so you need to pass in credentials in headers.

**Review by tochman — COMMENTED** [review]
I would like you to revisit this before merge.

**Review by Fleurtam — APPROVED** [review]
looks good!

File: config/routes.rb (hunk 2 above)
Comment (AKidd95) [reply config/routes.rb]: Ok, this is good to know, When I tried to change it I got an `uninitialized constant API::V1:UserControlller` error. even after changing the name of the controller

**Review by isabellelii — APPROVED** [review]

File: spec/requests/api/v1/user_request_spec.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+      expected_response = eval(file_fixture('users.txt').read)
+    end
+
+    it 'Should return one user' do
+      get '/api/v1/user/:id', headers: {
+        email: user.email, password: user.password
+      }
+
+      expected_response = eval(file_fixture('user.txt').read)
```
Comment (holgertidemand) [inline spec/requests/api/v1/user_request_spec.rb:21]: You are actually not testing anything here. 
You need to add something like this after this line.
``` 
expect(object['data'].first).to eq expected_response
``` 
this just reads the user.txt file and sets expected_response equal to it :)
```
expected_response = eval(file_fixture('user.txt').read)
```

Just a heads up.

**Review by holgertidemand — COMMENTED** [review]
Just a heads up :)

**Pull request merged**